storage/ndb/rest-server/rest-api-server/internal/handlers/feature_store/handler.go

```diff
@@ -0,0 +1,358 @@
+/*
+ * This file is part of the RonDB REST API Server
+ * Copyright (c) 2023 Hopsworks AB
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package feature_store
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"hopsworks.ai/rdrs/internal/config"
+	"hopsworks.ai/rdrs/internal/feature_store"
+	"hopsworks.ai/rdrs/internal/handlers/batchpkread"
+	"hopsworks.ai/rdrs/internal/log"
+	"hopsworks.ai/rdrs/internal/security/apikey"
+	"hopsworks.ai/rdrs/pkg/api"
+)
+
+const (
+	JSON_NUMBER  = "NUMBER"
+	JSON_STRING  = "STRING"
+	JSON_BOOLEAN = "BOOLEAN"
+	JSON_NIL     = "NIL"
+	JSON_OTHER   = "OTHER"
+)
+
+type Handler struct {
+	fvMetaCache   *feature_store.FeatureViewMetaDataCache
+	apiKeyCache   apikey.Cache
+	dbBatchReader batchpkread.Handler
+}
+
+func New(fvMetaCache *feature_store.FeatureViewMetaDataCache, apiKeyCache apikey.Cache, batchPkReadHandler batchpkread.Handler) Handler {
+	return Handler{fvMetaCache, apiKeyCache, batchPkReadHandler}
+}
+
+func (h *Handler) Validate(request interface{}) error {
+	fsReq := request.(*api.FeatureStoreRequest)
+	metadata, err := h.fvMetaCache.Get(
+		*fsReq.FeatureStoreName, *fsReq.FeatureViewName, *fsReq.FeatureViewVersion)
+	if err != nil {
+		return err.GetError()
+	}
+	if log.IsDebug() {
+		log.Debugf("Feature store request is %s", fsReq.String())
+	}
+	err1 := ValidatePrimaryKey(fsReq.Entries, &metadata.PrefixFeaturesLookup)
+	if err1 != nil {
+		return err1.GetError()
+	}
+	err2 := ValidatePassedFeatures(fsReq.PassedFeatures, &metadata.PrefixFeaturesLookup)
+	if err2 != nil {
+		return err2.GetError()
+	}
+	return nil
+
+}
+
+func ValidatePrimaryKey(entries *map[string]*json.RawMessage, features *map[string]*feature_store.FeatureMetadata) *feature_store.RestErrorCode {
+	// Data type check of primary key will be delegated to rondb.
+	if len(*entries) == 0 {
```

Comment:
It is not suppose to be nil. But the pointer is coming from `Entries` in `FeatureStoreRequest`

Reply:
Yeah, so essentially from

```go
postParams := api.FeatureStoreRequest{}
if err := c.BindJSON(&postParams); err != nil {
	return nil, err
}
```
So if Entries does not exist in the JSON, this will be nil?

You could use GIN's `binding:"required"` required syntax, have a look at the `BatchResponseJSON` struct.